src/uu/split/src/number.rs

```diff
@@ -0,0 +1,513 @@
+    /// See the [`DynamicWidthNumber`] documentation for more
+    /// information.
+    ///
+    /// # Errors
+    ///
+    /// This method returns [`Err(Overflow)`] when attempting to
+    /// increment beyond the largest representable number.
+    ///
+    /// # Examples
+    ///
+    /// Overflowing:
+    ///
+    /// ```rust,ignore
+    ///
+    /// use crate::number::FixedWidthNumber;
+    /// use crate::number::Number;
+    /// use crate::number::Overflow;
+    ///
+    /// // Radix 3, width of 1 digit.
+    /// let mut number = Number::FixedWidth(FixedWidthNumber::new(3, 1));
+    /// number.increment().unwrap();  // from 0 to 1
+    /// number.increment().unwrap();  // from 1 to 2
+    /// assert!(number.increment().is_err());
+    /// ```
+    pub fn increment(&mut self) -> Result<(), Overflow> {
+        match self {
+            Number::FixedWidth(number) => number.increment(),
+            Number::DynamicWidth(number) => number.increment(),
+        }
+    }
+}
+
+impl Display for Number {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Number::FixedWidth(number) => number.fmt(f),
+            Number::DynamicWidth(number) => number.fmt(f),
+        }
+    }
+}
+
+/// A positional notation representation of a fixed-width number.
+///
+/// The digits are represented as a [`Vec<u8>`] with the most
+/// significant digit on the left and the least significant digit on
+/// the right. Each digit is a nonnegative integer less than the
+/// radix.
+///
+/// # Incrementing
+///
+/// This number starts at `vec![0; width]`, representing the number 0
+/// width the specified number of digits. Incrementing this number
+/// with [`increment`] causes it to increase its value by 1 in the
+/// usual sense. If the digits are `vec![radix - 1; width]`, then an
+/// overflow would occur and the [`increment`] method returns an
+/// error.
+///
+/// # Displaying
+///
+/// This number is only displayable if `radix` is 10 or `radix` is
+/// 26. If `radix` is 10, then the digits are concatenated and
+/// displayed as a fixed-width decimal number. If `radix` is 26, then
+/// each digit is translated to the corresponding lowercase ASCII
+/// alphabetic character (that is, 'a', 'b', 'c', etc.) and
+/// concatenated.
+#[derive(Clone)]
+pub struct FixedWidthNumber {
+    radix: u8,
+    digits: Vec<u8>,
+}
+
+impl FixedWidthNumber {
+    /// Instantiate a number of the given radix and width.
+    pub fn new(radix: u8, width: usize) -> FixedWidthNumber {
+        FixedWidthNumber {
+            radix,
+            digits: vec![0; width],
+        }
+    }
+
+    /// Increment this number.
+    ///
+    /// This method adds one to this number. If incrementing this
+    /// number would require more digits than are available with the
+    /// specified width, then this method returns [`Err(Overflow)`].
+    fn increment(&mut self) -> Result<(), Overflow> {
+        for i in (0..self.digits.len()).rev() {
```

Comment:
To increment the number, add 1 to each digit from least significant digit to most significant digit, carrying the 1 if necessary. An overflow results in an error for `FixedWidthNumber`.